src/main/java/de/bayern/gdi/gui/Controller.java

```diff
@@ -598,8 +598,15 @@ protected void handleSaveConfig(ActionEvent event) {
             }
 
             FileChooser fileChooser = new FileChooser();
+            FileChooser.ExtensionFilter xmlFilter =
+                    new FileChooser.ExtensionFilter("xml files (*.xml)", "xml");
+            fileChooser.getExtensionFilters().add(xmlFilter);
+            fileChooser.setSelectedExtensionFilter(xmlFilter);
             fileChooser.setTitle(I18n.getMsg("gui.save-conf"));
             File configFile = fileChooser.showSaveDialog(getPrimaryStage());
+            if (!configFile.toString().endsWith(".xml")) {
+                configFile = new File(configFile.toString() + ".xml");
+            }
```

Comment:
Not the right direction. Think about overwriting an existing file. If you enter a name that exists and acknowledge to overwrite it, this mechanism would overwrite an other file.
A better way would be to suggest a file with an .xml extension like fileChooser.setInitialFileName("config.xml") before calling the dialog!?